framework/src/functions/MooseParsedFunctionWrapper.C

```diff
@@ -121,7 +121,18 @@ MooseParsedFunctionWrapper::initialize()
     // Case when a Real is supplied
     else
     {
-      Real val = MooseUtils::convert<Real>(_vals_input[i], true);
+      Real val;
+      try
+      {
+        val = MooseUtils::convert<Real>(_vals_input[i], true);
+      }
+      catch (std::exception & e)
+      {
+        mooseError("No postprocessor, scalar variable, or function with the name '",
```

Comment:
Can you add the function block name so its obvious?
```
mooseError("In ", _name, ": blah blah...")
```

Reply:
Actually in the wrapper class I don't have access to the name.

Reply:
Passing in the name would require an API change. Let me see how much that would break...

Reply:
Well, at least it errors now...but sometimes it's a pain to find where it errors. It'd be nice to make it as easy as possible. Maybe you could just print the equation/value?

Reply:
I could catch the exception higher up, but then I don't have the vals item name...
Catch and rethrow with a modified message would work. But this is becoming ugly...

Reply:
It's probably fine since you print out the offending name